internal/pkg/peer/blocksprovider/blocksprovider.go

```diff
@@ -285,6 +238,163 @@ func (d *Deliverer) DeliverBlocks() {
 	}
 }
 
+func (d *Deliverer) DeliverBlocksNew() {
+	if d.BlockGossipDisabled {
+		d.Logger.Infof("Will pull blocks without forwarding them to remote peers via gossip")
+	}
+
+	fc := &replication.FetcherConfig{ // TODO these value need to be pulled from config
+		Channel:                      d.ChannelID,
+		PeriodicalShuffleInterval:    200 * time.Second,
+		CensorshipSuspicionThreshold: 10 * time.Second,
+		FetchTimeout:                 20 * time.Second,
+		RetryTimeout:                 5 * time.Millisecond,
+		MaxRetries:                   100,
+		BufferSize:                   2097152,
+		TLSCertHash:                  d.TLSCertHash,
+	}
+
+	d.Logger.Warnf("**** d: %+v", d)
+
+	config := d.ConfigProvider.Resources()
+
+	blockSigVerifierFactory := &replication.BlockVerifierAssembler{
+		Logger: d.Logger,
+		BCCSP:  d.BCCSP,
+	}
+	var blockSigVerifier replication.BlockVerifier
+
+	verifyBlockSequence := func(blocks []*common.Block, channel string) error {
+		// TODO some of this is probably redundant
+		if len(blocks) == 0 {
+			return errors.New("buffer is empty")
+		}
+		if blocks[0] == nil {
+			return errors.New("first block is nil")
+		}
+		if blocks[0].Header == nil {
+			return errors.New("first block header is nil")
+		}
+		if blocks[0].Header.Number == 0 {
+			configEnv, err := replication.ConfigFromBlock(blocks[0])
+			if err != nil {
+				return errors.WithMessage(err, "failed to extract config envelope from genesis block")
+			}
+			// Bootstrap the verifier from the genesis block, as it will be used to verify
```

Comment:
I understand you copied this code, but we should never need to pull the genesis block, and also we should never blindly accept a genesis block to be used for initialization of any validation logic.